sys/fmt/fmt.c

```diff
@@ -472,6 +473,19 @@ size_t fmt_to_lower(char *out, const char *str)
     return len;
 }
 
+int fmt_time_tm_iso8601(char *out, const struct tm *tm, char separator)
```

Comment:
```suggestion
int fmt_time_tm_iso8601(char out[20], const struct tm *tm, char separator)
```

Wouldn't this work? I think this is currently the best way to inform the compiler on how much data may be written to a buffer.

With this, passing a too short buffer should give a warning